src/ext/java/org/opentripplanner/ext/siri/updater/AsyncEstimatedTimetableProcessor.java

```diff
@@ -0,0 +1,51 @@
+package org.opentripplanner.ext.siri.updater;
+
+import java.util.concurrent.ExecutionException;
+import uk.org.siri.siri20.ServiceDelivery;
+
+/**
+ * Process a SIRI-ET feed by combining an asynchronous source of estimated timetables
+ * {@link AsyncEstimatedTimetableSource} with a consumer of estimated timetables
+ * {@link EstimatedTimetableHandler}
+ */
+public class AsyncEstimatedTimetableProcessor {
+
+  private final AsyncEstimatedTimetableSource siriMessageSource;
+  private final EstimatedTimetableHandler estimatedTimetableHandler;
+
+  private volatile boolean primed;
+
+  public AsyncEstimatedTimetableProcessor(
+    AsyncEstimatedTimetableSource siriMessageSource,
+    EstimatedTimetableHandler estimatedTimetableHandler
+  ) {
+    this.siriMessageSource = siriMessageSource;
+    this.estimatedTimetableHandler = estimatedTimetableHandler;
+  }
+
+  public void run() {
+    siriMessageSource.start(this::processSiriData);
+  }
+
+  public boolean isPrimed() {
+    return this.primed;
+  }
+
+  private void processSiriData(ServiceDelivery serviceDelivery) {
+    var f = estimatedTimetableHandler.applyUpdate(
+      serviceDelivery.getEstimatedTimetableDeliveries(),
+      false
+    );
+    if (!isPrimed()) {
```

Comment:
Will this priming logic be general enough for the current use cases? It looks pretty specific.

At least for the azure updaters the priming is done separately from the main update logic.

Reply:
This will cause the updater to be marked as primed as soon as the message backlog (history) is applied.
In the Azure case, the updater is primed after the message backlog is applied and the event processor is started.
So indeed this is not exactly equivalent.
I can probably refactor this to make it more flexible.

Reply:
I moved the priming logic into  `AsyncEstimatedTimetableSource` since this is where we have better control on when the source is actually ready.
I also slighly modified the logic in the PubSub updater to mark it primed after the message consumer is started (that is: the same logic as in the Azure updater).

Another difference is that the updater thread in the PubSub case blocks forever (`subscriber.awaitTerminated()`), while in the Azure case the thread stops immediately after startup (`org.opentripplanner.ext.siri.updater.azure.AbstractAzureSiriUpdater#run` is non-blocking)
As far as I can see, this is not a problem.

Reply:
I think having the priming logic in the source is a good choice. And I can't see any issue with having the updater thread blocking vs exiting either.